Sources/MuxUploadSDK/PublicAPI/Options/DirectUploadOptions.swift

```diff
@@ -29,6 +29,25 @@ public struct DirectUploadOptions {
             )
         }
 
+        /// Initializes options for upload chunk transport
+        /// over the network
+        /// - Parameters:
+        ///     - chunkSize: the size of each file chunk sent
+        ///     by the SDK during an upload.
+        ///     Defaults to 8MB.
+        ///     - retryLimitPerChunk: number of retry attempts
+        ///     if the chunk request fails, default value is 3
+        public init(
+            chunkSize: Measurement<UnitInformationStorage> = .defaultDirectUploadChunkSize,
+            retryLimitPerChunk: Int = 3
+        ) {
+            self.chunkSizeInBytes = Int(
+                abs(chunkSize.converted(to: .bytes).value)
```

Comment:
In Foundation quantities of bytes get represented as a [`Double`](https://developer.apple.com/documentation/foundation/measurement/1924053-value) which makes me wonder if we should too.  As a counterargument we'd still need to do rounding when we use `Progress` though.

Reply:
Seems weird to me but, if it makes sense in this context then 👍  🍫

Reply:
Let's think about it more and make any API adjustments from `Int` to `Double` in a separate PR